src/util/key.rs

```diff
@@ -11,86 +11,94 @@
 // If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 //
 
-//! Private key
+//! Bitcoin Keys
 //!
-//! A private key represents the secret data associated with its proposed use
+//! Structures and methods representing the public/secret data associated with
+//! its proposed use
 //!
 
 use std::fmt::{self, Write};
 use std::str::FromStr;
 use secp256k1::{self, Secp256k1};
-use secp256k1::key::{PublicKey, SecretKey};
 use util::address::Address;
 use consensus::encode;
 use network::constants::Network;
 use util::base58;
 
-#[derive(Clone, PartialEq, Eq)]
-/// A Bitcoin ECDSA private key
-pub struct Privkey {
-    /// Whether this private key represents a compressed address
+/// A Bitcoin ECDSA public key
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct PublicKey {
+    /// Whether this public key represents a compressed address
```

Comment:
Isn't it kinda weird to think of a public key to represent an address instead of the other way around? I realize this is not a change, but it's the first time I come across this code.

Reply:
Yes, the word "represents" should be replaced with "is part of" or something like that.